packages/lesswrong/server/repos/PageCacheRepo.ts

```diff
@@ -0,0 +1,38 @@
+import type { CompleteTestGroupAllocation } from "../../lib/abTestImpl";
+import PageCache from "../../lib/collections/pagecache/collection";
+import { getClientBundle } from "../utils/bundleUtils";
+import AbstractRepo from "./AbstractRepo";
+
+export type MeanPostKarma = {
+  _id: number,
+  meanKarma: number,
+}
+
+export default class PageCacheRepo extends AbstractRepo<DbPageCacheEntry> {
+  constructor() {
+    super(PageCache);
+  }
+
+  async lookupCacheEntry({path, completeAbTestGroups}: {path: string, completeAbTestGroups: CompleteTestGroupAllocation}): Promise<DbPageCacheEntry | null> {
+    const { bundleHash } = getClientBundle();
+
+    // Note: we use db.any here rather than e.g. oneOrNone because there may (in principle) be multiple
+    // abTestGroups in the db that are a subset of the completeAbTestGroups. In this case it shouldn't
+    // matter which one we use, so we just take the first one.
+    const cacheResult = await this.getRawDb().any(`
+      SELECT * FROM "PageCache"
+      WHERE "path" = $1
+      AND "bundleHash" = $2
+      AND "expiresAt" > NOW()
+      AND jsonb_subset($3::jsonb, "abTestGroups")`,
```

Comment:
Minor note: I've been prepending all of our user-defined function names with `fm_` to make it obvious that they are in fact user-defined. My first though when I saw this was that `jsonb_subset` was some built-in Postgres thing that I wasn't aware of.

This may not actually be necessary, but now seems like a reasonable time to make a decision about whether this should be something we do or not.

(I've not reviewed the rest of the PR).

Reply:
I like this convention and also would have thought `jsonb_subset` was a postgres builtin